include/RevOpts.h

```diff
@@ -18,15 +18,46 @@
 #include <cinttypes>
 #include <map>
 #include <string>
+#include <tuple>
+#include <type_traits>
 #include <utility>
 #include <vector>
 
 namespace SST::RevCPU {
 
 class RevOpts {
+
+  // Queries whether type is a std::vector type
+  template<typename T>
+  struct is_vector : std::false_type {};
+
+  template<typename T>
+  struct is_vector<std::vector<T>> : std::true_type {};
```

Comment:
This is standard traits programming, where a trait query class is defined as false in general, but then a _partial specialization_ is defined for which it's true for a proper subset of the original template.